src/taoensso/sente/server_adapters/undertow.clj

```diff
@@ -40,24 +42,28 @@
   (read!  [this])
   (close! [this]))
 
-(deftype SenteUndertowAjaxChannel [ch open?_ on-close]
+(deftype SenteUndertowAjaxChannel [promised-response open?_ on-close]
   ISenteUndertowAjaxChannel
-  (send!  [this msg] (async/put! ch msg (fn [_] (close! this))))
-  (read!  [this] (async/<!! ch))
+  (send!  [this msg] (deliver promised-response msg))
+  (read!  [this]
+    (let [resp (deref promised-response ajax-response-timeout-ms :lp-timed-out)]
```

Comment:
Two changes requested here:
1. For compatibility with the old behaviour, please make the timeout optional.
2. Suggest the timeout value be configurable via `get-sch-adapter` option. Maybe a default value of `:undertow/ajax-response-timeout`? (`ajax-response-timeout-val` to better match the `:ajax-response-timeout-ms` option name, and the `undertow` prefix to help clarify that this is an adapter-specific keyword).